File: leetcode/137.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int ret = 0;
        int mask = 1;
        for(int i = 0; i < 32; i++)
        {
            int num = 0;
            for(int j = 0; j < nums.size(); j++)
            {
                if (nums[j] & mask)
                    num++;
            }
            num %= 3;
            mask <<= 1;
            ret |= (num << i);
        }
        return ret;
    }
};


int main()
{
    return 0;
}

File: leetcode/137.py
#!/usr/bin/env python
# -*- coding:utf-8 -*-
# date:20160901

class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        nums_dic = collections.Counter(nums)
        for i in nums_dic:
            if nums_dic[i] == 1:
                return i